### PR TITLE
recursion-impl

### PR DESCRIPTION
**Pull request by PaulFrmBrn**
Section 1 / recursion
Implementation

### CODE REVIEW

File: recursion/src/main/java/Factorial.java [hunk 1]
```diff
@@ -1,6 +1,12 @@
 public class Factorial {
 
     int of(int base) {
-        throw new UnsupportedOperationException();
+        if (base <= 0) {
```
Comment (zamonier) [inline recursion/src/main/java/Factorial.java]: Uncovered statement

Comment (PaulFrmBrn) [reply recursion/src/main/java/Factorial.java]: I'll name new PR to fix tests

Comment (PaulFrmBrn) [reply recursion/src/main/java/Factorial.java]: done - https://github.com/PaulFrmBrn/udemy-ds-and-algo-course/pull/7

Comment (PaulFrmBrn) [reply recursion/src/main/java/Factorial.java]: this branch was also rebased

**Pull request merged**